students/preactive/Lesson4/MailroomPt2.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python3
+import sys
+from pathlib import Path
+import pickle
+
+
+donorsHistory = Path("donors.pkl")
+if donorsHistory.is_file():
+    donersFile = open(r'donors.pkl', 'rb')
+    donors = pickle.load(donersFile)
+    donersFile.close()
+else:
+    donors = [
+        {'donor':'Hutch', 'amounts':[10, 15, 75.75] },
+        {'donor':'Carlton', 'amounts':[32, 19000.01]},
+        {'donor':'William Gates, III', 'amounts':[1000000, 17000000, 3.99]},
+        ]
+    donorsFile = open(r'donors.pkl', 'wb')
+    pickle.dump(donors, donorsFile)
+    donorsFile.close()
+
+def userPrompt():
+    choice = input(
+        'Pick one:\n1) Send a thank you\n2) Create a Report\n3) Quit\n>')
+    while choice not in ['1', '2', '3']:
+        choice = input(
+            'Pick one:\n1) Send a thank you\n2) Create a Report\n 3) Quit\n>')
+    return main(choice)
```

Comment:
combine your `main` with this function to simplify, example:
```
def exit_program():
    print("Exiting")
    sys.exit()

options = {"1": send_thank_you, "2": create_report, "3": exit_program}
while True:
    selection = input("Please select .. ")
    options[selection]()
```

this is done once, and send_thank_you and other function should have knowledge of this CLI flow.